### PR TITLE
Change to classes

### PR DESCRIPTION
**Pull request by dmytro-lt**
https://app.asana.com/0/1157793729216339/1201246276117418

### CODE REVIEW

File: src/twig.compiler.js [hunk 1]
```diff
@@ -10,40 +10,85 @@ export default function (Twig) {
...
+    //         output = Twig.compiler.module[options.module](id, tokens, options.twig);
+    //     } else {
+    //         output = Twig.compiler.wrap(id, tokens);
+    //     }
+
+    //     return output;
+    // };
+    class TwigCompilerComplite {
+
+        Twig;
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js]: Why is  this need?

File: src/twig.compiler.js [hunk 2]
```diff
@@ -10,40 +10,85 @@ export default function (Twig) {
...
+    //     },
+    //     node(id, tokens) {
+    //         return 'var twig = require("twig").twig;\nexports.template = ' + Twig.compiler.wrap(id, tokens);
+    //     },
+    //     cjs2(id, tokens, pathToTwig) {
+    //         return 'module.declare([{ twig: "' + pathToTwig + '" }], function (require, exports, module) {\n\tvar twig = require("twig").twig;\n\texports.template = ' + Twig.compiler.wrap(id, tokens) + '\n});';
+    //     }
+    // };
+
+    class TwigCompilerModule {
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js]: It mus be class field

File: src/twig.compiler.js [hunk 3]
```diff
@@ -10,40 +10,85 @@ export default function (Twig) {
...
-        return 'twig({id:"' + id.replace('"', '\\"') + '", data:' + tokens + ', precompiled: true});\n';
-    };
+    class TwigCompilerWrap {
+        fn(id, tokens) {
+            return 'twig({id:"' + id.replace('"', '\\"') + '", data:' + tokens + ', precompiled: true});\n';
+        }
+
+        // return Twig
+    }
+    // Twig.compiler.wrap = function (id, tokens) {
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js]: Method of class

**Review by andriy-kozlytski — CHANGES_REQUESTED** [review]

File: src/twig.compiler.js [hunk 4]
```diff
@@ -8,42 +8,37 @@ export default function (Twig) {
...
-        if (options.module) {
-            if (Twig.compiler.module[options.module] === undefined) {
-                throw new Twig.Error('Unable to find module type ' + options.module);
+                output = Twig.compiler.module[options.module](id, tokens, options.twig);
+            } else {
+                output = Twig.compiler.wrap(id, tokens);
+            }
+            return output;
+        } 
+        module() {
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js]: It isn't method it's field

File: src/twig.compiler.js [hunk 5]
```diff
@@ -8,42 +8,37 @@ export default function (Twig) {
...
-    };
-
-    Twig.compiler.wrap = function (id, tokens) {
-        return 'twig({id:"' + id.replace('"', '\\"') + '", data:' + tokens + ', precompiled: true});\n';
-    };
-
-    return Twig;
-};
+    }
+    return Twig
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js]: shouldn't return

File: src/twig.compiler.js [hunk 6]
```diff
@@ -8,42 +8,37 @@ export default function (Twig) {
     Twig.compiler = {
         module: {}
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js]: this is field of TwigCompiler

**Review by andriy-kozlytski — CHANGES_REQUESTED** [review]
Should export class

File: src/twig.compiler.js [hunk 7]
```diff
@@ -8,42 +8,37 @@ export default function (Twig) {
...
-        amd(id, tokens, pathToTwig) {
-            return 'define(["' + pathToTwig + '"], function (Twig) {\n\tvar twig, templates;\ntwig = Twig.twig;\ntemplates = ' + Twig.compiler.wrap(id, tokens) + '\n\treturn templates;\n});';
-        },
-        node(id, tokens) {
-            return 'var twig = require("twig").twig;\nexports.template = ' + Twig.compiler.wrap(id, tokens);
-        },
-        cjs2(id, tokens, pathToTwig) {
-            return 'module.declare([{ twig: "' + pathToTwig + '" }], function (require, exports, module) {\n\tvar twig = require("twig").twig;\n\texports.template = ' + Twig.compiler.wrap(id, tokens) + '\n});';
+        wrap() {
+            fn: (id, tokens) => 'twig({id:"' + id.replace('"', '\\"') + '", data:' + tokens + ', precompiled: true});\n';
```
Comment (lodry) [inline src/twig.compiler.js]: why is it object?

**Review by lodry — CHANGES_REQUESTED** [review]

File: src/twig.compiler.js [hunk 8]
```diff
@@ -25,25 +22,12 @@ export default function (Twig) {
         } else {
             output = Twig.compiler.wrap(id, tokens);
```
Comment (lodry) [inline src/twig.compiler.js]: it is not injected in oyur class better `this.wrap(id, tokens)`

File: src/twig.compiler.js [hunk 9]
```diff
@@ -1,16 +1,13 @@
...
 //
 // This file handles compiling templates into JS
-export default function (Twig) {
-    /**
-     * Namespace for compilation.
-     */
-    Twig.compiler = {
+export default class TwigCompiler {
+
+    constructor() {
```
Comment (lodry) [inline src/twig.compiler.js]: inject global Twig object

File: src/twig.compiler.js [hunk 10]
```diff
@@ -1,16 +1,13 @@
 // ## twig.compiler.js
 //
 // This file handles compiling templates into JS
-export default function (Twig) {
-    /**
-     * Namespace for compilation.
-     */
-    Twig.compiler = {
+export default class TwigCompiler {
```
Comment (lodry) [inline src/twig.compiler.js]: you should export not class but instantiated object

File: src/twig.compiler.js [hunk 11]
```diff
@@ -25,25 +22,12 @@ export default function (Twig) {
...
-            return 'define(["' + pathToTwig + '"], function (Twig) {\n\tvar twig, templates;\ntwig = Twig.twig;\ntemplates = ' + Twig.compiler.wrap(id, tokens) + '\n\treturn templates;\n});';
-        },
-        node(id, tokens) {
-            return 'var twig = require("twig").twig;\nexports.template = ' + Twig.compiler.wrap(id, tokens);
-        },
-        cjs2(id, tokens, pathToTwig) {
-            return 'module.declare([{ twig: "' + pathToTwig + '" }], function (require, exports, module) {\n\tvar twig = require("twig").twig;\n\texports.template = ' + Twig.compiler.wrap(id, tokens) + '\n});';
-        }
-    };
+    module: {}
```
Comment (lodry) [inline src/twig.compiler.js]: why is it empty??

**Review by lodry — CHANGES_REQUESTED** [review]

File: src/twig.compiler.js [hunk 12]
```diff
@@ -1,49 +1,43 @@
...
+    compile(template, options) {
         // Get tokens
         const tokens = JSON.stringify(template.tokens);
         const {id} = template;
         let output = null;
 
         if (options.module) {
-            if (Twig.compiler.module[options.module] === undefined) {
-                throw new Twig.Error('Unable to find module type ' + options.module);
+            if (this.module[options.module] === undefined) {
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js:17]: replace to
this.Twig.compiler.

File: src/twig.compiler.js [hunk 13]
```diff
@@ -1,49 +1,43 @@
...
+                throw new this.Twig.Error('Unable to find module type ' + options.module);
             }
 
-            output = Twig.compiler.module[options.module](id, tokens, options.twig);
+            output = this.module[options.module](id, tokens, options.twig);
         } else {
-            output = Twig.compiler.wrap(id, tokens);
+            output = this.wrap(id, tokens);
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js:23]: same

File: src/twig.compiler.js [hunk 14]
```diff
@@ -1,49 +1,43 @@
...
 
         if (options.module) {
-            if (Twig.compiler.module[options.module] === undefined) {
-                throw new Twig.Error('Unable to find module type ' + options.module);
+            if (this.module[options.module] === undefined) {
+                throw new this.Twig.Error('Unable to find module type ' + options.module);
             }
 
-            output = Twig.compiler.module[options.module](id, tokens, options.twig);
+            output = this.module[options.module](id, tokens, options.twig);
```
Comment (andriy-kozlytski) [inline src/twig.compiler.js:21]: same

**Review by andriy-kozlytski — CHANGES_REQUESTED** [review]

**Review by lodry — APPROVED** [review]

**Pull request merged**